tests/Console/Command/ListFilesCommandTest.php

```diff
@@ -41,6 +41,34 @@ public function testListWithConfig(): void
         self::assertSame(escapeshellarg($expectedPath).PHP_EOL, $commandTester->getDisplay());
     }
 
+    /**
+     * @requires OS Linux|Darwin
+     *
+     * Skip test on Windows as `getcwd()` includes the drive letter with a colon `:` which is illegal in filenames.
+     */
+    public function testListFilesDoesNotCorruptListWithGetcwdInName(): void
+    {
+        $cwd = getcwd();
+        self::assertIsString($cwd, 'Cannot get the current working directory.');
+
+        $tempFile = __DIR__.'/../../Fixtures/ListFilesTest/using-getcwd/'.ltrim($cwd, '/').'-out.php';
```

Comment:
I don't like this idea - it creates a lot of nested directories in the `Fixtures` and since there is no any `.gitignore` rule for that, in case of failure it may lead to situation when you have some temp file for commit. Maybe we can utilise `sys_get_temp_dir()` and `chdir()` to create files in temp dir, then re-use that path within subdirectory? There are already tests in the project that leave some mess, I would like to avoid new ones 😅.

<img width="1415" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/8731a5ea-4424-4d17-a798-e0f2dd3e03c1">

Reply:
Very annoying to test, indeed. I reverted using `sys_get_temp_dir()` as I don't know what's happening on macOS. Anyway, I think I used proper post-cleanup so I think this should be okay now.